### ISSUE TITLE
Resterende arbejde på prototype til at oprette tidsserieskridt til nivellementtidsserie

### ISSUE THREAD

**Issue by xidus** [Kortforsyningen/FIRE#553]
|             Opgave             |                                           Hvad                                           | Estimat / h |
|--------------------------------|------------------------------------------------------------------------------------------|-------------|
| Beregn netværk                 | Giv inputdata til `gama_beregn`                                                          |        2.00 |
| Kontrol af GAMA-filer          | Se på GAMA-inddata og -uddata                                                            |        2.00 |
| Føj resultater til tidsserie   | Find ud af, hvordan de skal tilføjes tidsserien                                          |        4.00 |
| Plot resultater                | Timebox                                                                                  |        0.50 |
| Læg resultater i databasen     |                                                                                          |        4.00 |
| Hent tidsserien ud igen        | Anvend eksisterende funktionalitet og kontrollér indsættelse                             |        0.25 |
| Anvend ny datamodel            | Erstat metoder i `fire.io.ts` med kald til ORM-hentere-ilæggere |        3.00 |
| 5D-punkter                     |                                                                                          |        9999 |
| Brugertests                    |                                                                                          |        2.00 |
| Tilpasninger efter brugertests |                                                                                          |        9999 |

**Comment by kbevers** [comment]
En væsentligt opgave der også mangler er rent faktisk at populere de nye `tidsserie` og `punktsamling` tabeller med information. Der skal defineres en mapping fra Koordinater med SRID TS:81xxx til Tidsserie og PunktSamling og diverse forbindelser til Koordinat, Punkt og SRID. Måske kan det klares fuldautomatisk med ren SQL, måske er det semi-manuel proces der skal klares for hver jessenpunkt+tilhørende tidsserier. Tiden må vise hvad der er den bedste løsning.

**Comment by xidus** [comment]
Ja. Og ud over brugertest/sparring er skal der inden da også være implementeret test-funktionalitet:)

**Comment by xidus** [comment]
Der er væsentlige udfordringer med at få anvendt `gama_beregning` til formålet her:

* Vores præmis er, at der ikke er behov for et projekt-regneark på dén lokation, man kører koden fra.
* Funktionen kører GAMA, men ud over dette, laver den en opdatering af dén punktoversigt (`arbejdssæt`), den blev givet som input, hvilket betyder, at denne Pandas DataFrame skal fyldes med flere data for at kunne lave ændringerne.

Da vi ikke laver en sagsbehandling på samme måde som målerne, men i stedet bare laver en enkelt beregning af et netværk, hvor vi selv sikrer os at de rette forudsætninger for dette er opfyldt, så bliver det unødigt kompliceret at bruge funktionen, hvis ikke den refaktoriseres.

Denne refaktorisering tager ekstra tid.

**Comment by kbevers** [comment]
Det kunne godt lyde som om der er behov for en generisk funktion til at skrive en input-fil til gama, som så kan erstatte noget af indmaden i `gama_beregning` og samtidigt understøtte den nye use-case med tidsserie.

**Comment by xidus** [comment]
Præcis, jeg har givet et bud på det med seneste commits (67201a), hvor det bliver redelegeret til brugeren/programmet, hvordan man oversætter sine data til noget, GNU Gama kan bruge, og tilbage til dét format, man skal bruge i den givne situation.

Jeg har nu kode, der kan lave en Gama-beregning og returnere en liste med noget, der kan læses ind som tidsserieskridt.

Der mangler stadig noget, der kan læse de beregnede data ind som tidsserieskridt i databasen.

Men skal vi tage et skridt tilbage til start og se, hvordan datastrømmen skal være?

Indtil videre bruger jeg stadig rå SQL til at hente data ud, men hvis det er muligt at bruge en ny datamodel, så giver det måske mening at tage denne i brug med det samme.

Ellers kan det rettes til senere.

Indtil videre regner jeg med at fortsætte med test og ilægning af data.

**Comment by kbevers** [comment]
> Men skal vi tage et skridt tilbage til start og se, hvordan datastrømmen skal være?
>
> Indtil videre bruger jeg stadig rå SQL til at hente data ud, men hvis det er muligt at bruge en ny datamodel, så giver det måske mening at tage denne i brug med det samme.

Jeg synes det giver bedst mening at tage de nye tidsserie- og punktsamlingsobjekter i brug med det samme. Vi kan lige så godt få trykprøvet konceptet og jeg forventer også du sparer en del tid ved at gå den vej frem for håndholdt SQL.

**Comment by xidus** [comment]
OK, jeg bruger ORMen.

**Comment by kbevers** [comment]
Arbejde med højdetidsserier implementeret i https://github.com/SDFIdk/FIRE/pull/779
